File: Java/TestCreditCardValidator.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class TestCreditCardValidator {
	CreditCardValidator ccv = new CreditCardValidator();
	
	@Test 
	public void testCheckTypeFunctionExists() {
		String entry = "4388576018410707";
		int length = entry.length();
		assertNotNull(ccv.checkType(entry, length));
	}
	
	@Test	
	public void testCheckTypeFunctionAmericanExpress() {
		String entry = "373233232323432";
		int length = entry.length();
		String actual = ccv.checkType(entry, length);
		String expected = "American Express Card"; 
		assertEquals(actual, expected);
	}	
		
	@Test
	public void testCheckTypeFunctionVisa() {
		String entry = "4388576018402626";
		int length = entry.length();
		String actual = ccv.checkType(entry, length);
		String expected = "Visa"; 
		assertEquals(actual, expected);
	}	
	
	@Test	
	public void testCheckTypeFunctionMasterCard() {
		String entry = "5388586018702826";
		int length = entry.length();
		String actual = ccv.checkType(entry, length);
		String expected = "MasterCard"; 
		assertEquals(actual, expected);
	}	
	
	@Test	
	public void testCheckTypeFunctionDiscover() {
		String entry = "6388676018402628";
		int length = entry.length();
		String actual = ccv.checkType(entry, length);
		String expected = "Discover";
		assertEquals(actual, expected);
	}	
	
	@Test	
	public void testCheckTypeFunctionInvalidCard() {
		String entry = "2388576018402626";
		int length = entry.length();
		String actual = ccv.checkType(entry, length);
		String expected = "Invalid Card"; 
		assertEquals(actual, expected);
	}
	
	@Test	
	public void testCheckValidityFunctionExists() {
		String entry = "4388576018410707";
		assertNotNull(ccv.checkValidity(entry));
	}
	
	@Test	
	public void testCheckValidityFunctionForValid() {
		String entry = "4388576018402626";
		String actual = ccv.checkValidity(entry);
		String expected = "Valid";
		assertEquals(actual, expected);
	}
	
	@Test	
	public void testCheckValidityFunctionForInvalid() {
		String entry = "4388576018410707";
		String actual = ccv.checkValidity(entry);
		String expected = "Invalid";
		assertEquals(actual, expected);
	}
}

File: Python/testcreditcardvalidator.py
import creditcardvalidator

from unittest import TestCase

class TestCreditCardValidator(TestCase):

	def test_check_type_function_exists(self):
		entry = "4388576018410707"
		length = len(entry)
		self.assertIsNotNone(creditcardvalidator.check_type(entry, length))
		
	def test_check_type_function_american_express(self):
		entry = "373233232323432"
		length = len(entry)
		actual = creditcardvalidator.check_type(entry, length)
		expected = "American Express Card" 
		self.assertEqual(actual, expected)	
		
	def test_check_type_function_visa(self):
		entry = "4388576018402626"
		length = len(entry)
		actual = creditcardvalidator.check_type(entry, length)
		expected = "Visa" 
		self.assertEqual(actual, expected)	
		
	def test_check_type_function_mastercard(self):
		entry = "5388586018702826"
		length = len(entry)
		actual = creditcardvalidator.check_type(entry, length)
		expected = "MasterCard" 
		self.assertEqual(actual, expected)	
		
	def test_check_type_function_discover(self):
		entry = "6388676018402628"
		length = len(entry)
		actual = creditcardvalidator.check_type(entry, length)
		expected = "Discover" 
		self.assertEqual(actual, expected)	
		
	def test_check_type_function_visa(self):
		entry = "2388576018402626"
		length = len(entry)
		actual = creditcardvalidator.check_type(entry, length)
		expected = "Invalid Card" 
		self.assertEqual(actual, expected)
		
	def test_check_validity_function_exists(self):
		entry = "4388576018410707"
		self.assertIsNotNone(creditcardvalidator.check_validity(entry))
		
	def test_check_validity_function_for_valid(self):
		entry = "4388576018402626"
		actual = creditcardvalidator.check_validity(entry)
		expected = "Valid"
		self.assertEqual(actual, expected)
		
	def test_check_validity_function_for_invalid(self):
		entry = "4388576018410707"
		actual = creditcardvalidator.check_validity(entry)
		expected = "Invalid"
		self.assertEqual(actual, expected)
		